pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -4499,21 +4507,29 @@ func (c *Cloud) getInstancesByNodeNames(nodeNames []string, states ...string) ([
 
 // TODO: Move to instanceCache
 func (c *Cloud) describeInstances(filters []*ec2.Filter) ([]*ec2.Instance, error) {
+	// Set the Maxresults to the maximum allowed per the AWS doc
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeInstancesInput
+	maxResults := int64(1000)
 	request := &ec2.DescribeInstancesInput{
-		Filters: filters,
+		Filters:    filters,
+		MaxResults: &maxResults,
 	}
 
-	response, err := c.ec2.DescribeInstances(request)
+	var matches []*ec2.Instance
+	err := c.ec2.DescribeInstancesPages(request, func(page *ec2.DescribeInstancesOutput, lastPage bool) bool {
+		for _, reservations := range page.Reservations {
+			for _, instance := range reservations.Instances {
+				if c.tagging.hasClusterTag(instance.Tags) {
+					matches = append(matches, instance)
+				}
+			}
+		}
+		return !lastPage
```

Comment:
Would it be clearer to just return `true`?

Reply:
I was not sure about what the behaviour would be if I return true there on the last page.